backend/dataall/modules/datasets_base/db/dataset_repositories.py

```diff
@@ -290,6 +290,27 @@ def query_environment_datasets(session, uri, filter) -> Query:
             )
         return query
 
+    @staticmethod
+    def query_environment_imported_datasets(session, uri, filter) -> Query:
```

Comment:
Is there a reason we limit the restriction of S3 permissions to be only for imported datasets and for data.all created datasets we still have more general s3 permissions on a wildcard `resource_prefix` S3 ARN?

Have you looked into limiting s3 permissions to only the explicit set of s3 bucket ARNs that are mapped to data.all "Datasets" (either imported or created) or is there a risk of quickly reaching the policy size limit with that approach?

Reply:
I was worried about the policy size limitations, so to be optimal I only add the imported datasets to that policy. For the data.all created datasets there is already a policy that grants all those permissions. In `backend/dataall/core/environment/cdk/pivot_role_core_policies/s3.py` we are granting permissions to all data.all managed buckets. Besides optimizing the size of the policies and the amount of policies, I wanted to reduce the amount of times the pivot role is updated. It should not give any issues, but less CloudFormation operations are less opportunities for errors

```
            # S3 Managed Buckets
            iam.PolicyStatement(
                sid='ManagedBuckets',
                effect=iam.Effect.ALLOW,
                actions=[
                    's3:List*',
                    's3:Delete*',
                    's3:Get*',
                    's3:Put*'
                ],
                resources=[f'arn:aws:s3:::{self.env_resource_prefix}*'],
            ),
```